src/plugins/formulas/formulas.js

```diff
@@ -522,6 +536,45 @@ export class Formulas extends BasePlugin {
     return value;
   }
 
+  /**
+   * `onBeforeAutofill` hook callback.
+   *
+   * @private
+   * @param {Array[]} fillData The data that was used to fill the `targetRange`. If `beforeAutofill` was used
+   * and returned `[[]]`, this will be the same object that was returned from `beforeAutofill`.
+   * @param {CellRange} sourceRange The range values will be filled from.
+   * @param {CellRange} targetRange The range new values will be filled into.
+   * @returns {boolean|*}
+   */
+  onBeforeAutofill(fillData, sourceRange, targetRange) {
+    const sourceLeftCorner = { ...sourceRange.from, sheet: this.sheetId };
+    const sourceWidth = sourceRange.getWidth();
+    const sourceHeight = sourceRange.getHeight();
+    const targetLeftCorner = { ...targetRange.from, sheet: this.sheetId };
+    const targetWidth = targetRange.getWidth();
+    const targetHeight = targetRange.getHeight();
+
+    // Blocks the autofill operation if HyperFormula says that at least one of the underlying's cell contents cannot be set.
+    if (this.engine.isItPossibleToSetCellContents(targetLeftCorner, targetWidth, targetHeight) === false) {
+      return false;
+    }
+
+    const sourceEnd = {
+      row: sourceLeftCorner.row + sourceHeight - 1,
+      col: sourceLeftCorner.col + sourceWidth - 1,
+      sheet: sourceLeftCorner.sheet,
+    };
+    const source = { start: sourceLeftCorner, end: sourceEnd };
+    const targetEnd = {
+      row: targetLeftCorner.row + targetHeight - 1,
+      col: targetLeftCorner.col + targetWidth - 1,
+      sheet: targetLeftCorner.sheet,
+    };
+    const target = { start: targetLeftCorner, end: targetEnd };
+
+    return this.engine.getFillRangeData(source, target);
```

Comment:
1. The `isItPossibleToSetCellContents` currently only takes 1 argument (a single cell address or a cell range): https://handsontable.github.io/hyperformula/api/classes/hyperformula.html#isitpossibletosetcellcontents 
2. I think the code may be simplified a little bit, as the `CellRange` class contains methods like `getTopLeftCorner` and `getBottomRightCorner`, so unless I'm missing something, there's no need for retrieving the width/height of the ranges and manually calculating the ends of ranges.

Maybe something like this could also work?

```suggestion
    const extendWithSheetId = range => Object.assign(range, { sheet: this.sheetId });
    const engineSourceRange = {
      start: extendWithSheetId(sourceRange.getTopLeftCorner()),
      end: extendWithSheetId(sourceRange.getBottomRightCorner())
    };
    const engineTargetRange = {
      start: extendWithSheetId(targetRange.getTopLeftCorner()),
      end: extendWithSheetId(targetRange.getBottomRightCorner())
    };

    // Blocks the autofill operation if HyperFormula says that at least one of
    // the underlying cell's contents cannot be set.
    if (this.engine.isItPossibleToSetCellContents(engineTargetRange) === false) {
      return false;
    }

    return this.engine.getFillRangeData(engineSourceRange, engineTargetRange);
```

Reply:
Seems to work great, thanks 👍 https://github.com/handsontable/handsontable/pull/8669/commits/fb6bd9eb0b257aa22a2f646d79c6a375ede19868